check_for_messages.py

```diff
@@ -53,20 +54,212 @@ def find_subreddit(words):
         if word == afterWord:
             return words[i]
 
-def process_rbot_message(bot_id, words):
-    subreddit = ''
+def find_word_before(words, beforeWord):
+    i = 0
+    for word in words:
+        if word == beforeWord:
+            return words[i-1]
+        i += 1
+
+def set_auto_interval(interval, words):
+    int(interval)
+    if interval != 0:
+        if "hour" in words:
+            interval = interval * 60 * 60
+        elif "hours" in words:
+            interval = interval * 60 * 60
+        elif "half" in words:
+            interval = interval * 30 * 60
+        elif "minutes" in words:
+            interval = interval * 60
+	    elif "minute" in words:
+	        interval = interval *60
+    elif interval == 0:
+        if "hours" in words:
+            interval = 120 * 60
+        elif "hour" in words:
+            interval = 60 * 60
+        elif "half" in words:
+            interval = 30 * 60
+        elif "minutes" in words:
+	        interval = find_word_before(words, "minutes")
+	        if is_number(interval):
+	            int(interval)
+                interval = interval * 60
+	        else:
+                interval = 5 * 60
+	    elif "minute" in words:
+	        interval = 60
+
+    return interval
+
+def remove_auto_params(bot_id):
+    try:
+        keystore.delete('auto_sub')
+        keystore.delete('auto_interval')
+        keystore.delete('last_auto_post')
+        keystore.delete('auto_count')
+    except:
+        return
+
+    lastChange = time.time()
+    keystore.put('last_auto_change', lastChange)
+    keystore.put('active_auto', 0)
+    groupme.post_message(bot_id, "Auto posting stopped")
+    return
+
+
+def check_subreddit(bot_id, subreddit):
+    subredditResult = reddit.subreddit_exists( subreddit )
+
+    if subredditResult.error is not None:
+        groupme.post_message( bot_id, subredditResult.error )
+        return False
+
+    if not subredditResult.exists:
+        message = "Subreddit /r/{0} does not exist.".format(subreddit)
+        if subredditResult.suggestion is not None:
+            message += " Did you mean /r/{0}?".format(subredditResult.suggestion)
+        groupme.post_message(bot_id, message)
+        return False
+
+    return True
+
+def check_blacklist(bot_id, subreddit):
+    blacklisted_subs_file = "blacklisted_subs.csv"
+    if os.path.isfile(blacklisted_subs_file):
+        with open(blacklisted_subs_file, 'rb') as csvfile:
+            reader = csv.reader(csvfile, delimiter=',')
+            for row in reader:
+                if subreddit.lower() in (sub.lower() for sub in row):
+                    groupme.post_message(bot_id, "We're not gonna go there")
+                    return True
+    return False
+
+def check_active_auto():
+    auto = 0
+    try:
+	    auto = int(keystore.get('active_auto'))
+    except:
+	    return False
+    if auto == 1:
+	    return True
+    else:
+	    return False
+
+def check_auto_words(bot_id, subreddit, words):
```

Comment:
this methods pretty long, with lots of nested if statements. Maybe the "change" code could be put in its own method, and the "every" code